yarn-project/simulator/src/avm/avm_machine_state.ts

```diff
@@ -46,6 +46,7 @@ export class AvmMachineState {
   constructor(gasLeftOrL2GasLeft: Gas | number, daGasLeft?: number) {
     if (typeof gasLeftOrL2GasLeft === 'object') {
       ({ l2Gas: this.l2GasLeft, daGas: this.daGasLeft } = gasLeftOrL2GasLeft);
+      this.l2GasLeft -= FIXED_AVM_STARTUP_L2_GAS;
```

Comment:
Are you aware making this change here in the machine state will also charge the cost for each external call (even each nested)? (even down the road when we do 1 enqueued call per AVM circuit)

Reply:
Yes, that was the intent, because as it stands we need to generate a public kernel proof for each external/nested call right? And that is what this startup cost is primarily intended to cover.